src/Observable.ts

```diff
@@ -37,6 +37,10 @@ import {ErrorObservable} from './observable/throw';
 import {AjaxCreationMethod} from './observable/dom/ajax';
 import {WebSocketSubject} from './observable/dom/webSocket';
 
+export type ObservableOrPromise<T> = Observable<T> | Promise<T>;
```

Comment:
I can't think of an operator that doesn't have to import Observable, because all operators I can think of, return observables.  That's kind of why I liked the types in here, it's less work on the operator side to bring them in and use them.